packages/odyssey-react-mui/src/createMessageBus.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+export type MessageEvent<Message> = (message: Message) => void;
+
+export type PublishMessage<Message> = MessageEvent<Message>;
+
+export type UnsubscribeMessageSubscription = () => void;
+
+export type MessageSubscription<Message> = (
+  subscription: MessageEvent<Message>,
+) => UnsubscribeMessageSubscription;
+
+export type MessageBus<Message> = {
+  publish: PublishMessage<Message>;
+  subscribe: MessageSubscription<Message>;
+};
+
+export const createMessageBus = <Message = void>(): MessageBus<Message> => {
+  const subscribers = new Map<symbol, MessageEvent<Message>>();
+
+  const publish: PublishMessage<Message> = (message) => {
+    Array.from(subscribers.values()).forEach((subscriber) => {
+      subscriber(message);
+    });
+  };
+
+  const subscribe: MessageSubscription<Message> = (subscription) => {
+    const subscriptionId = Symbol();
+    subscribers.set(subscriptionId, subscription);
+
+    return () => {
+      subscribers.delete(subscriptionId);
+    };
+  };
+
+  return {
+    publish,
+    subscribe,
+  };
+};
```

Comment:
`createMessageBus` exists because it provides the cleanest API to consumers of Odyssey. I tried a bunch of different methods, and this gave us the most versatility and the easiest ease-of-use.

I can simply return `publish`; no one needs to setup their own event bus.